xds/src/main/java/io/grpc/xds/AbstractXdsClient.java

```diff
@@ -72,6 +73,18 @@
   private static final String ADS_TYPE_URL_EDS =
       "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
 
+  private final SynchronizationContext syncContext = new SynchronizationContext(
+      new Thread.UncaughtExceptionHandler() {
+        @Override
+        public void uncaughtException(Thread t, Throwable e) {
+          getLogger().log(
+              XdsLogLevel.ERROR,
```

Comment:
Can we include some identifying info for which XdsClient was logging this: ClientXdsClient or ServerXdsClient and for both what resources were being watched/handled.

Reply:
At current stage, there will be only one XdsClient in the process. I don't think we are ready to have both client and server running in the same process now.

Reply:
I think your comments need some clarification: there will be only one ClientXdsClient in the process but one ServerXdsClient per gRPC server (listener). Also " I don't think we are ready to have both client and server running in the same process now." is definitely not true. What you probably meant is not ready to have a single XdsClient to handle both client side and server side. 

The identifying information would still be helpful on the server side because of one XdsClient per server.

Reply:
But each XdsClient instance already has a unique logId, which is included in XdsLogger. Log messages already include information for which XdsClient it is logged from.